File: 0118-pascals-triangle/0118-pascals-triangle.js
var generate = function(numRows) {
    var i = 0;
    var j = 0;
    // Create an array list to store the output result...
    var res = [];
    // For generating each row of the triangle...
    for (i = 0; i < numRows; i++) {
        res.push(Array(i + 1));       // Initialize the first row of the pascal triangle as {1}...
        for (j = 0; j <= i; j++) {
            // Primary...
            if (j === 0 || j === i) {
                res[i][j] = 1;
            }
            else {
                // Calculate the elements of a row, add each pair of adjacent elements of the previous row in each step of the inner loop.
                res[i][j] = res[i - 1][j - 1] + res[i - 1][j];
            }
        }
    }
    return res;      // Return the output list of pascal values...
};

File: 0118-pascals-triangle/0118-pascals-triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        
        
        List<List<Integer>> triangle = new ArrayList<>();
        if(numRows == 0){
            return triangle;
        }
        
        List<Integer> first_row = new ArrayList<>();
        first_row.add(1);
        
        triangle.add(first_row);
        
        for(int row = 1; row < numRows; row++){
            
            List<Integer> prevRow = triangle.get(row - 1);
            List<Integer> newRow = new ArrayList<>();
            newRow.add(1);
            
            for(int col = 1; col <= row -1; col++){
                newRow.add(prevRow.get(col -1) + prevRow.get(col));
            }
            
            newRow.add(1);
            triangle.add(newRow);
        }
        return triangle;
    }
}